### PR TITLE
python312Packages.inflect: 7.2.1 -> 7.3.1

### PR DESCRIPTION
**Pull request by dotlambda**
## Description of changes
Diff: https://github.com/jaraco/inflect/compare/refs/tags/v7.2.1...v7.3.1

Changelog: https://github.com/jaraco/inflect/blob/v7.3.1/CHANGES.rst

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Review by jnsgruk — APPROVED** [review]

File: pkgs/development/python-modules/inflect/default.nix [hunk 1]
```diff
@@ -1,31 +1,32 @@
...
+  disabled = pythonOlder "3.8";
+
+  src = fetchFromGitHub {
+    owner = "jaraco";
+    repo = "inflect";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-J0XgSKPzZIt/7WnMGARXpyYzagBGiqRiuNmNnGKDBrs=";
   };
 
-  nativeBuildInputs = [ setuptools-scm ];
```
Comment (drupol) [inline pkgs/development/python-modules/inflect/default.nix:24]: I guess we also have to add `setuptools` in here. I'll do it.

Comment (dotlambda) [reply pkgs/development/python-modules/inflect/default.nix:24]: No.

Comment (drupol) [reply pkgs/development/python-modules/inflect/default.nix:24]: => https://github.com/NixOS/nixpkgs/pull/329849#discussion_r1691063277

Comment (drupol) [reply pkgs/development/python-modules/inflect/default.nix:24]: @dotlambda @natsukium Please advise ^^

Comment (natsukium) [reply pkgs/development/python-modules/inflect/default.nix:24]: We should not depend on implicit propagation.

Comment (dotlambda) [reply pkgs/development/python-modules/inflect/default.nix:24]: What does @mweinelt think?

Comment (drupol) [reply pkgs/development/python-modules/inflect/default.nix:24]: Don't worry, it has been merged already: https://github.com/NixOS/nixpkgs/pull/330114

Comment (natsukium) [reply pkgs/development/python-modules/inflect/default.nix:24]: Please stop the pointless discussion.
https://github.com/NixOS/nixpkgs/pull/265993/files#r1385614792

Comment (dotlambda) [reply pkgs/development/python-modules/inflect/default.nix:24]: I just want to come to a consensus. But given I'm outvoted I'm happy to add setuptools in the future.

Comment (mweinelt) [reply pkgs/development/python-modules/inflect/default.nix:24]: I don't care.

**Pull request merged**